src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryDate.cs

```diff
@@ -301,17 +301,18 @@ public static FormulaValue DateDiff(EvalVisitor runner, EvalVisitorContext conte
                     return new NumberValue(irContext, years);
             }
 
-            // Convert to UTC to be accurate (apply DST if needed)
             var timeZoneInfo = runner.GetService<TimeZoneInfo>() ?? LocalTimeZone;
 
-            if (NeedToConvertToUtc(runner, start, timeUnit))
+            // Subtract an hour if time falls in DST
+            if (timeZoneInfo.IsDaylightSavingTime(start))
             {
-                start = TimeZoneInfo.ConvertTimeToUtc(start, timeZoneInfo);
+                start = start.AddHours(-1);
```

Comment:
WE should avoid -1 for DST, and instead rely on ConvertTimeToUtc

Reply:
Agree, Revisiting the logic for it.